### DISCUSSION TITLE
Project governance?

### DISCUSSION THREAD

**Discussion by mattdm** [StractOrg/stract#150] [General]
The CLA simply says 'Stract ("We" or "Us")', and doesn't explain what that is.

I know this is a young project, but what is the intended structure? Who makes decisions — and who ultimately owns the code? When there are disputes, how will they be handled? 

Individual "benevolent dictator for life" can be a fine structure (at least at first), but it would be nice to know. There's no need to over-complicate things — I'd just appreciate clarity. 

Thanks!

**Comment by mikkeldenker** [comment]
Hi!

The CLA should really say 'Stract ApS'. It's a company I have created here in Denmark that is currently fully owned by me and my family. This is the entity that ultimately owns the code.

I think it would be better to view Stract as an open source company. There have been multiple incredible projects in the past who have tried to take a more donation driven approach to search ([common search](https://github.com/commonsearch), [alexandria](https://github.com/alexandria-org/alexandria), and more recently [mwmbl](https://github.com/mwmbl/mwmbl) and [marginalia](https://github.com/marginaliasearch/marginaliasearch)). Building a search engine, much less keeping it running, is by no means a trivial task and I personally don't think a fully donation driven approach is sustainable long term. I might very well be wrong here, and I really hope these projects succeed, but this is at least the reason why Stract tries to take a different approach.

Being open source ensures that Stract's incentives will always be aligned with the user. If we ever did anything that's not in the users best interest, like changing the license, someone would simply take our code and start a competitor and we would lose all our users. That's why Stract will never do something like that.

The CLA is put in place to make it [easier for Stract to sue a competitor](https://softwareengineering.stackexchange.com/a/168026) if they choose to take our code without making their code available, which would be a violation of the agpl license. This is ultimately good for the user.

As you said, stract is still a *very* young project and I will do my best to communicate these things better. I'll try to make it more clear in the `CONTRIBUTING.md` file to hopefully avoid any misunderstandings.

I fully understand if someone doesn't want to contribute under this structure, that's totally fine. I wouldn't want anyone to contribute unless they feel comfortable doing so. I will still continue to make all my work available here for everyone.
